packages/@ama-sdk/core/src/plugins/timeout/timeout.fetch.ts

```diff
@@ -1,51 +1,120 @@
 import { ResponseTimeoutError } from '../../fwk/errors';
 import { FetchCall, FetchPlugin, FetchPluginContext } from '../core';
 
+/**
+ * Representation of an Imperva Captcha message
+ */
+type ImpervaCaptchaMessageData = {
+  impervaChallenge: {
+    type: 'captcha';
+    status: 'started' | 'ended';
+    timeout: number;
+    url: string;
+  };
+};
+
+
+/**
+ * Check if a message can be cast as an {@link ImpervaCaptchaMessage}
+ * @param message
+ */
+function isImpervaCaptchaMessage(message: any): message is ImpervaCaptchaMessageData {
+  return Object.prototype.hasOwnProperty.call(message, 'impervaChallenge') &&
+    Object.prototype.hasOwnProperty.call(message.impervaChallenge, 'status') &&
+    Object.prototype.hasOwnProperty.call(message.impervaChallenge, 'type') && message.impervaChallenge.type === 'captcha';
+}
+
+export type TimeoutPauseEventHandler = ((timeoutPauseCallback: (isTimeoutPaused: boolean) => void, context: any) => () => void);
```

Comment:
Since this behaviour is not necessary for the feature and that the current behaviour has been tested for the captcha feature. I see how we could benefit from this - however would it be possible to do it in a second time and adapt the code to make the handler a configuration to avoid future breaking change ?